tutorials/euclid_access/4_Euclid_intro_PHZ_catalog.md

```diff
@@ -190,53 +189,47 @@ im_cutout= 5 * u.arcmin
 ra_cutout = 267.8
 dec_cutout =  66
 
-coords_cutout = SkyCoord(ra_cutout, dec_cutout, unit=(u.deg, u.deg), frame='icrs')
-##########################################################################
-im_cutout_deg=im_cutout.to(u.deg).value
-
-## Create ra and dec bounds for the box
-ra0=ra_cutout-im_cutout_deg/2
-ra1=ra_cutout+im_cutout_deg/2
-
-dec0=dec_cutout-im_cutout_deg/2
-dec1=dec_cutout+im_cutout_deg/2
+coords_cutout = SkyCoord(ra_cutout, dec_cutout, unit='deg', frame='icrs')
+size_cutout = im_cutout.to(u.deg).value
 ```
 
 ```{code-cell} ipython3
-adql = f"SELECT DISTINCT mer.object_id,mer.ra, mer.dec, phz.flux_vis_unif, phz.flux_y_unif, \
-phz.flux_j_unif, phz.flux_h_unif, phz.phz_classification, phz.phz_median, phz.phz_90_int1, phz.phz_90_int2 \
-FROM {table_mer} AS mer \
-JOIN {table_phz} as phz \
-ON mer.object_id = phz.object_id \
-WHERE 1 = CONTAINS(POINT('ICRS', mer.ra, mer.dec), CIRCLE('ICRS', {ra_cutout}, {dec_cutout}, {im_cutout_deg/2})) \
-AND phz.flux_vis_unif> 0 \
-AND  phz.flux_y_unif > 0 \
-AND phz.flux_j_unif > 0 \
-AND phz.flux_h_unif > 0 \
-AND phz.phz_classification = 2 \
-AND ((phz.phz_90_int2 - phz.phz_90_int1) / (1 + phz.phz_median)) < 0.20 \
-AND phz.phz_median BETWEEN 1.4 AND 1.6 \
-"
-adql
+adql = ("SELECT DISTINCT mer.object_id, mer.ra, mer.dec, "
+        "phz.flux_vis_unif, phz.flux_y_unif, phz.flux_j_unif, phz.flux_h_unif, "
+        "phz.phz_classification, phz.phz_median, phz.phz_90_int1, phz.phz_90_int2 "
+        f"FROM {table_mer} AS mer "
+        f"JOIN {table_phz} as phz "
+        "ON mer.object_id = phz.object_id "
+        "WHERE 1 = CONTAINS(POINT('ICRS', mer.ra, mer.dec), "
+                            f"BOX('ICRS', {ra_cutout}, {dec_cutout}, {size_cutout/np.cos(coords_cutout.dec)}, {size_cutout})) "
+        "AND phz.flux_vis_unif> 0 "
+        "AND  phz.flux_y_unif > 0 "
+        "AND phz.flux_j_unif > 0 "
+        "AND phz.flux_h_unif > 0 "
+        "AND phz.phz_classification = 2 "
+        "AND ((phz.phz_90_int2 - phz.phz_90_int1) / (1 + phz.phz_median)) < 0.20 "
+        "AND phz.phz_median BETWEEN 1.4 AND 1.6")
 
 
 ## Use TAP with this ADQL string
-result = Irsa.query_tap(adql)
-
+result_galaxies = Irsa.query_tap(adql).to_table()
+result_galaxies[:5]
+```
 
-## Convert table to pandas dataframe
-df_g_irsa = result.to_table().to_pandas()
+```{warning}
+Note that we use `to_table` above rather than `to_qtable`. While astropy's `QTable` is more powerful than its `Table`, as it e.g. handles the column units properly, we cannot use it here due to a known bug; it mishandles the large integer numbers in the `object_id` column and recast them as float during which process some precision is being lost. 
```

Comment:
I added a note about the bug as I already forgot about it and changed this line to use `to_qtable` which messes up the object_id.

Fix will come in the DB probably quicker than in astropy; so we can cleanup the warning and the code once those updates are out (not using `NA` as unit for the `long` columns).